src/jdk.jdeps/share/classes/com/sun/tools/jnativescan/JNativeScanTask.java

```diff
@@ -0,0 +1,193 @@
+        while (!classPathJars.isEmpty()) {
+            Path jar = classPathJars.poll();
+            checkRegularJar(jar);
+            String[] classPathAttribute = classPathAttribute(jar);
+            Path parentDir = jar.getParent();
+            for (String classPathEntry : classPathAttribute) {
+                Path otherJar = parentDir != null
+                        ? parentDir.resolve(classPathEntry)
+                        : Path.of(classPathEntry);
+                if (Files.exists(otherJar)) {
+                    // Class-Path attribute specifies that jars that
+                    // are not found are simply ignored. Do the same here
+                    classPathJars.offer(otherJar);
+                }
+            }
+            builder.add(new ScannedModule(jar, "ALL-UNNAMED"));
+        }
+        return builder.build();
+    }
+
+    private String[] classPathAttribute(Path jar) {
+        try (JarFile jf = new JarFile(jar.toFile(), false, ZipFile.OPEN_READ, version)) {
+           Manifest manifest = jf.getManifest();
+           if (manifest != null) {
+               String attrib = manifest.getMainAttributes().getValue("Class-Path");
+               if (attrib != null) {
+                   return attrib.split("\\s+");
+               }
+           }
+           return new String[0];
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private Configuration systemConfiguration() {
+        ModuleFinder systemFinder = ModuleFinder.ofSystem();
+        Configuration system = Configuration.resolve(systemFinder, List.of(Configuration.empty()), ModuleFinder.of(),
+                allModuleNames(systemFinder)); // resolve all of them
+        return system;
+    }
+
+    private List<String> allModuleNames(ModuleFinder finder) {
+        return finder.findAll().stream().map(mr -> mr.descriptor().name()).toList();
+    }
+
+    private void printNativeAccess(Map<ScannedModule, Map<ClassDesc, List<RestrictedUse>>> allRestrictedMethods) {
+        String nativeAccess = allRestrictedMethods.keySet().stream()
+                .map(ScannedModule::moduleName)
+                .distinct()
+                .collect(Collectors.joining(","));
+        out.println(nativeAccess);
+    }
+
+    private void dumpAll(Map<ScannedModule, Map<ClassDesc, List<RestrictedUse>>> allRestrictedMethods) {
+        if (allRestrictedMethods.isEmpty()) {
+            out.println("  <no restricted methods>");
+        } else {
+            allRestrictedMethods.forEach((module, perClass) -> {
+                out.println(module.path() + " (" + module.moduleName() + "):");
+                perClass.forEach((classDesc, restrictedUses) -> {
+                    out.println("  " + qualName(classDesc) + ":");
+                    restrictedUses.forEach(use -> {
+                        switch (use) {
+                            case RestrictedUse.NativeMethodDecl(MethodRef nmd) ->
```

Comment:
should user be able to filter output using an option? E.g. say if they are only interested in restricted methods but not JNI (seems remote, but adding it here for completeness)

Reply:
That's a good suggestion, but to pull on that string some more, there's also the possibility of filtering based on class/package/method names. Maybe for a followup?